manifests/artemis.pp

```diff
@@ -20,20 +22,56 @@
     group   => $candlepin::group,
   }
 
-  file { "${candlepin::catalina_home}/conf/cert-users.properties":
-    ensure  => file,
-    content => Deferred('inline_epp', ["katelloUser=<%= \$artemis_client_dn %>\n", {'artemis_client_dn' => $candlepin::artemis_client_dn}]),
-    mode    => '0640',
-    owner   => $candlepin::user,
-    group   => $candlepin::group,
+  if $candlepin::artemis_client_certificate_user_map {
+    file { $artemis_cert_dir:
+      ensure => directory,
+      owner  => $candlepin::user,
+      group  => $candlepin::group,
+    }
+
+    $candlepin::artemis_client_certificate_user_map.each |$certificate, $user| {
+      file { "${artemis_cert_dir}/${user}.crt":
+        ensure => file,
+        source => $certificate,
+        owner  => $candlepin::user,
+        group  => $candlepin::group,
+        mode   => '0640',
+      }
+    }
+
+    artemis_cert_users { "${candlepin::catalina_home}/conf/cert-users.properties":
+      ensure            => present,
+      certificate_users => $candlepin::artemis_client_certificate_user_map,
+      mode              => '0640',
+      owner             => $candlepin::user,
+      group             => $candlepin::group,
+    }
```

Comment:
I would have expected more of a concat style pattern rather than a single resource.

Reply:
That seems more complex, and I do not fully see the why. Maybe there's an easy way to use concat under the hood like we use File resources? They key part of using a provider here is that we need to calculate the DN *on* the agent and we need to do it in a way where we know all the input certificates are present.